test/unit/test_config.py

```diff
@@ -51,13 +50,6 @@ def test_config_eq():
     assert c_nested != c_nested_c2
 
 
-def test_config_no_self_attribute():
```

Comment:
What was the reason we had this constraint/test?

Reply:
I think we wanted to avoid nested/circular config dependencies that refer to itself, because the str/repr methods would create infinite strings. I don't think we need that since it never really happened that we accidentally made a config a child of one of its child configs.